### PR TITLE
Don't include all networks when browsing the room directory

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/9285

### CODE REVIEW

**Comment by jryans** [comment]
What's the significance of "when browsing the room directory"? Is `NetworkDropdown` used in other places?

Overall, I think a better description of the intent of this change would help review.

**Comment by ara4n** [comment]
to try to spec the intent of the behaviour (as i understand it):
 * by default, the matrix.org roomdir should show just Matrix rooms (not bridged rooms), under the [M] label for the server. This is to stop a random bridge (e.g. rizon) flooding the roomdir with massive random rooms, but instead keep the directory to Matrix native ones by default.
 * if you switch to the 'matrix.org' section, it should show all rooms mixed together.

In other words, this is how it worked for many years, and i'm trying to just get it to go back to that because it's very annoying to see obnoxious rooms from other networks being pulled into the top of the default room list just because some random room list happened to join them.

I'm failing to see if this PR achieves this.

**Comment by turt2live** [comment]
Without this PR we achieve what is described by Matthew.
